src/schedule/kat_key_schedule.rs

```diff
@@ -0,0 +1,274 @@
+        let group_id = randombytes(16);
+
+        let mut epochs = Vec::new();
+
+        // Generate info for all epochs
+        for epoch in 0..NUM_EPOCHS {
+            println!("Generating epoch: {:?}", epoch);
+            let (
+                confirmed_transcript_hash,
+                commit_secret,
+                joiner_secret,
+                welcome_secret,
+                epoch_secrets,
+                tree_hash,
+                group_context,
+                external_key_pair,
+            ) = generate(ciphersuite, &init_secret, &group_id, epoch);
+
+            let epoch_info = Epoch {
+                tree_hash: bytes_to_hex(&tree_hash),
+                commit_secret: bytes_to_hex(commit_secret.as_slice()),
+                psk_secret: bytes_to_hex(&[]), // XXX: not implemented yet.
+                confirmed_transcript_hash: bytes_to_hex(&confirmed_transcript_hash),
+                group_context: bytes_to_hex(group_context.serialized()),
+                joiner_secret: bytes_to_hex(joiner_secret.as_slice()),
+                welcome_secret: bytes_to_hex(welcome_secret.as_slice()),
+                init_secret: bytes_to_hex(epoch_secrets.init_secret().unwrap().as_slice()),
+                sender_data_secret: bytes_to_hex(epoch_secrets.sender_data_secret().as_slice()),
+                encryption_secret: bytes_to_hex(epoch_secrets.encryption_secret().as_slice()),
+                exporter_secret: bytes_to_hex(epoch_secrets.exporter_secret().as_slice()),
+                authentication_secret: bytes_to_hex(
+                    epoch_secrets.authentication_secret().as_slice(),
+                ),
+                external_secret: bytes_to_hex(epoch_secrets.external_secret().as_slice()),
+                confirmation_key: bytes_to_hex(epoch_secrets.confirmation_key().as_slice()),
+                membership_key: bytes_to_hex(epoch_secrets.membership_key().as_slice()),
+                resumption_secret: bytes_to_hex(epoch_secrets.resumption_secret().as_slice()),
+                external_pub: bytes_to_hex(
+                    &external_key_pair.public_key().encode_detached().unwrap(),
+                ),
+            };
+            epochs.push(epoch_info);
+            init_secret = epoch_secrets.init_secret().unwrap().clone();
+        }
+
+        let test = KeyScheduleTestVector {
+            cipher_suite: ciphersuite.name() as u16,
+            group_id: bytes_to_hex(&group_id),
+            initial_init_secret: bytes_to_hex(initial_init_secret.as_slice()),
+            epochs,
+        };
+        tests.push(test);
+    }
+
+    write("test_vectors/kat_key_schedule_openmls-new.json", &tests);
+}
+
+#[test]
+fn run_test_vectors() {
+    let tests: Vec<KeyScheduleTestVector> = read("test_vectors/kat_key_schedule_openmls.json");
+
+    for test_vector in tests {
+        let ciphersuite =
+            CiphersuiteName::try_from(test_vector.cipher_suite).expect("Invalid ciphersuite");
```

Comment:
Wouldn't it make sense to fail gracefully here? I would imagine that in the future, there could be ciphersuites in the test vectors that we don't support. Or is this just for our own test vectors and not for interop?

Reply:
good point. Not here though. This always gets us the ciphersuite (unless its one that's not in the RFC, but we don't care about that case). I'll check for the supported cipher suite a line below. I'll do the same for the encryption test vectors.

Reply:
Good point indeed, we should expect all the ciphersuites that are mentioned in the spec to also appear in the test vectors and we don't support all of them right now (and probably not for a while).